programs/perpetuals/src/instructions/get_assets_under_management.rs

```diff
@@ -32,6 +35,7 @@ pub fn get_assets_under_management(
     _params: &GetAssetsUnderManagementParams,
 ) -> Result<u128> {
     ctx.accounts.pool.get_assets_under_management_usd(
+        AumCalcMode::Last,
```

Comment:
should we allow client to choose the mode so that it can calculate the correct lp price when user want to add or remove liquidity

Reply:
so, when someone deposits liquidity to the protocol, the amount of lp tokens received is calculated as:
```
lp_amount = lp_supply * token_amount_usd / pool_aum_usd
```
if pool has a token that is easy to manipulate, then pool_aum_usd can be lowered temporarily leading to more LP tokens minted to the attacker. If instead, we calculated `pool_aum_usd` using not the last price but **max**(last_price, avg_price) then the problem would be mitigated.
For the remove liquidity instruction, the opposite takes place:
```
amount_returned_to_user_usd = lp_amount * pool_aum_usd / lp_supply
```
The greater the `pool_aum_usd` the more tokens will be returned back to the user. So to mitigate the price manipulation attack, **min**(last_price, avg_price) is used to calculate `pool_aum_usd`. 
But when pool.aum_usd is stored for future reference, neither **max** nor **min** should be used but rather **last** as before.
Hence the AumCalcMode was introduced. The user shouldn't be selecting it, proper mode is selected depending on the context.

Reply:
i understand the implementation of max(last_price, avg_price) and min(last_price, avg_price). user will use max instead of last price to get lp tokens. 

During high volatility period, the max(last_price, avg_price) and last price might be different even for BTC and ETH, if the client always uses the last price to calculate the amount of lp tokens the user will get in the frontend, the user will always get lesser lp tokens than what he saw in the frontend.

Another question here, should we add a slippage in add and remove liquidity? Users will lose funds if the max(last_price, avg_price) being manipulated when he deposit funds and vice versa for remove liquidity

Reply:
good point, I'll add a getter to return amount of LP tokens the user would receive along with add liquidity fee (and similar for remove liquidity). And add a new parameter to add/remove liquidity - min_amount_out

Reply:
#44

Reply:
Thanks for the quick response, LGTM